js/main.js

```diff
@@ -127,19 +151,49 @@ document.addEventListener('DOMContentLoaded', function (event) {
   }
 });
 
+function hasFinishedProfile() {
+  if (data.profile.username !== '' && data.profile.avatarUrl !== '' && data.profile.fullName !== '' && data.profile.localStorage !== '') {
+    return true;
+  }
+}
 document.addEventListener('click', function (event) {
   if (event.target.tagName !== 'A') {
     return false;
   }
-  if (event.target.getAttribute('data-view') === 'profile') {
+  if (event.target.getAttribute('data-view') === 'profile' && hasFinishedProfile()) {
     swapView('profile');
   } else if (event.target.getAttribute('data-view') === 'edit-profile') {
     swapView('edit-profile');
+    document.querySelector('.view-profile').classList.add('hidden');
+  } else if (event.target.getAttribute('data-view') === 'entries' && hasFinishedProfile()) {
+    swapView('entries');
+  } else if (event.target.getAttribute('data-view') === 'create-entry' && hasFinishedProfile()) {
+    swapView('create-entry');
   }
 });
 
-var $modal = document.querySelector('#modal');
-document.querySelector('.x').addEventListener('click', function (event) {
-  $modal.classList.add('hidden');
-  document.querySelector('.container').classList.remove('modal-check');
+var $entryUrl = document.querySelector('.entry-image-url');
+var $entryTitle = document.querySelector('.entry-title');
+var $entryNotes = document.querySelector('.entry-notes');
+
+var $createEntryImage = document.querySelector('.create-entry-image');
+
+$entryUrl.addEventListener('input', function (event) {
+  $createEntryImage.setAttribute('src', event.target.value);
+});
+
+var newEntryData = {};
```

Comment:
You should do this inside of the function definition for your submit event handler, so that the object only exists while the function is currently running 👍